### PR TITLE
Adds pre- and post- customize commands

### PR DESCRIPTION
**Pull request by nakomis**
Required to support TOSCA pre- and post- customize commands for relationship operations. See http://docs.oasis-open.org/tosca/TOSCA-Simple-Profile-YAML/v1.0/csprd01/TOSCA-Simple-Profile-YAML-v1.0-csprd01.html#_Toc430015767

Will run VanillaWindowsProcessWinrmExitStatusLiveTest manually and @robertgmoss or I will update with result

### CODE REVIEW

**Comment by nakomis** [comment]
Ran `mvn clean install -PLive -Dtest=VanillaWindowsProcessWinrmExitStatusLiveTest` and all tests passed

File: software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessSshDriver.java [hunk 1]
```diff
@@ -310,6 +310,20 @@ public void runPostInstallCommand() {
...
+    public void runPreCustomizeCommand() {
+        if(Strings.isNonBlank(getEntity().getConfig(BrooklynConfigKeys.PRE_CUSTOMIZE_COMMAND))) {
+            execute(getEntity().getConfig(BrooklynConfigKeys.PRE_CUSTOMIZE_COMMAND), "running pre-customize commands");
+        }
+    }
+
+    @Override
+    public void runPostCustomizeCommand() {
+        if (Strings.isNonBlank(entity.getConfig(BrooklynConfigKeys.POST_CUSTOMIZE_COMMAND))) {
+            execute(ImmutableList.of(entity.getConfig(BrooklynConfigKeys.POST_CUSTOMIZE_COMMAND)), "running post-customize commands");
```
Comment (sjcorbett) [inline software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessSshDriver.java]: Why make a list in this method but not in `pre`?

File: software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessDriver.java [hunk 2]
```diff
@@ -147,8 +147,18 @@ public void start() {
             }});
 
             DynamicTasks.queue("customize", new Runnable() { public void run() {
-                waitForConfigKey(BrooklynConfigKeys.CUSTOMIZE_LATCH);
-                customize();
+                DynamicTasks.queue("pre-customize-command", new Runnable() { public void run() {
+                    runPreCustomizeCommand();
```
Comment (sjcorbett) [inline software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessDriver.java]: I'm not sure how I feel about only waiting for this customise latch in the main task. I think I would expect the latch to be obtained before `pre-customize-command` is run too.

Comment (nakomis) [inline software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessDriver.java]: This was done to mirror the behaviour of the install command (and latch)

File: software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessSshDriver.java (hunk 1 above)
Comment (nakomis) [inline software/base/src/main/java/org/apache/brooklyn/entity/software/base/AbstractSoftwareProcessSshDriver.java]: I'd copied the pre- and post- install command methods, and hadn't noticed that one is using a list and one wasn't. As the pre- and post- launch commands both use a list, I'll update this (and pre-install) to use a list

**Comment by nakomis** [comment]
PR comments addressed

**Comment by sjcorbett** [comment]
@nakomis Thanks.

**Pull request merged**